src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/UserServiceTest.kt

```diff
@@ -518,4 +522,81 @@ class UserServiceTest {
       assertThat(result).isInstanceOf(AuthorisableActionResult.NotFound::class.java)
     }
   }
+
+  @Nested
+  inner class UpdateUserRolesAndQualificationsFromApiById {
+    private val mockHttpAuthService = mockk<HttpAuthService>()
+    private val mockOffenderService = mockk<OffenderService>()
+    private val mockCommunityApiClient = mockk<CommunityApiClient>()
+    private val mockUserRepository = mockk<UserRepository>()
+    private val mockUserRoleAssignmentRepository = mockk<UserRoleAssignmentRepository>()
+    private val mockUserQualificationAssignmentRepository = mockk<UserQualificationAssignmentRepository>()
+    private val mockProbationRegionRepository = mockk<ProbationRegionRepository>()
+    private val mockProbationAreaProbationRegionMappingRepository = mockk<ProbationAreaProbationRegionMappingRepository>()
+
+//    private val userService = spyk<UserService>(recordPrivateCalls = true)
+
+    private val userService = mockk<UserService>()
+
+    private val id = UUID.fromString("21b61d19-3a96-4b88-8df9-a5e89bc6fe73")
+    private val username = "SOMEPERSON"
+    private val forename = "Jim"
+    private val surname = "Jimmerson"
+    private val staffIdentifier = 5678
```

Comment:
Do we need all this setup here, or can we rely on the defaults set by the factory?

Reply:
Yes we can rely on the defaults. Good shout.